rotkehlchen/api/rest.py

```diff
@@ -4993,6 +4993,22 @@ def add_tokens_to_spam(self, tokens: list[EvmToken]) -> Response:
                 )
                 AssetResolver.clean_memory_cache(token.identifier)
 
+                chain_balances = self.rotkehlchen.chains_aggregator.balances.get(
+                    chain=(blockchain := token.chain_id.to_blockchain()),
+                )
+                for balances in chain_balances.values():
+                    # those can only bee dicts because we mark as spam only evm tokens
+                    in_assets = balances.assets.pop(token, None)  # type: ignore
+                    in_liabilities = balances.liabilities.pop(token, None)  # type: ignore
+
+                    if in_assets is not None or in_liabilities is not None:
+                        log.debug(f'Flushing cache after setting {token} as spam')
+                        self.rotkehlchen.chains_aggregator.flush_cache('query_balances')
+                        self.rotkehlchen.chains_aggregator.flush_cache(
+                            name='query_balances',
+                            blockchain=blockchain,
+                        )
```

Comment:
is this not missing `self.flush_cache(f'query_{chain_key}_balances')`

Reply:
no, intentionally I don't want to clear it to prevent a query to the balances endpoint from querying the blockchain